ddl/ddl_tiflash_test.go

```diff
@@ -649,3 +661,122 @@ func TestAlterDatabaseBasic(t *testing.T) {
 	// There is less TiFlash store
 	tk.MustGetErrMsg("alter database tiflash_ddl set tiflash replica 3", "the tiflash replica count: 3 should be less than the total tiflash server count: 2")
 }
+
+func checkBatchPandingNum(t *testing.T, tkx *testkit.TestKit, level string, value string, ok bool) {
+	l := len(tkx.MustQuery(fmt.Sprintf("show %v variables where Variable_name='tidb_batch_pending_tiflash_count' and Value='%v'", level, value)).Rows())
+	if ok {
+		require.Equal(t, 1, l)
+	} else {
+		require.Equal(t, 0, l)
+	}
+}
+
+func TestTiFlashBatchAddVariables(t *testing.T) {
+	store, _, tear := createStoreWithoutMockTiFlash(t)
+	defer tear()
+
+	tk := testkit.NewTestKit(t, *store)
+	tk.MustExec("set SESSION tidb_batch_pending_tiflash_count=5")
+	tk.MustExec("set GLOBAL tidb_batch_pending_tiflash_count=6")
+
+	checkBatchPandingNum(t, tk, "session", "5", true)
+	checkBatchPandingNum(t, tk, "global", "6", true)
+	checkBatchPandingNum(t, tk, "global", "1.5", false)
+
+	tk.MustGetErrMsg("set GLOBAL tidb_batch_pending_tiflash_count=1.5", "[variable:1232]Incorrect argument type to variable 'tidb_batch_pending_tiflash_count'")
+	checkBatchPandingNum(t, tk, "global", "6", true)
+
+	tk2 := testkit.NewTestKit(t, *store)
+	checkBatchPandingNum(t, tk2, "session", "6", true)
+}
+
+func execWithTimeout(tk *testkit.TestKit, to time.Duration, sql string) (bool, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), to)
+	defer cancel()
+	doneCh := make(chan error, 1)
+
+	go func(c context.Context) {
+		stmts, _ := tk.Session().Parse(context.Background(), sql)
+		for _, stmt := range stmts {
+			_, err := tk.Session().ExecuteStmt(ctx, stmt)
+			doneCh <- err
+			return
+		}
+		doneCh <- nil
+	}(ctx)
+	select {
+	case e := <-doneCh:
+		return false, e
+	case <-ctx.Done():
+		return true, nil
+	}
+}
+
+func TestTiFlashBasicRateLimiter(t *testing.T) {
+	s, teardown := createTiFlashContext(t)
+	defer teardown()
+	tk := testkit.NewTestKit(t, s.store)
+
+	threshold := 2
+	tk.MustExec("drop database if exists tiflash_ddl_limit")
```

Comment:
No need for this.

Reply:
Do you mean the drop is not necessary?

Reply:
Yes.

Reply:
I may have later tests that also use this db

Reply:
It is in different store and tests

Reply:
get that, we are not using gocheck anymore..